pkg/controller/utils/component.go

```diff
@@ -155,17 +156,29 @@ func (c componentHandler) createOrUpdateObject(ctx context.Context, obj client.O
 				if err := c.client.Create(ctx, obj); err != nil {
```

Comment:
I couldn't add a comment on the line above but the Secret case called obj.SetResourceVersion("") so the Create would work.